### ISSUE TITLE
UtilityFunction shouldn't complain if `super()` is called

### ISSUE THREAD

**Issue by backus** [troessner/reek#761]
I think this is sort of a duplicate of https://github.com/troessner/reek/issues/39 except the "fix" for that issue only covered when zsuper is used.  I think this should cover `super()` too.

**Comment by troessner** [comment]
Unfortunately I don't follow - AFAIS we're handling all forms of `super` correctly.
Do you have an example?

**Comment by backus** [comment]
Should have included an example, sorry about that.  Often I'll use something like [concord](https://github.com/mbj/concord) to reduce boilerplate.  Usually this means not defining initialize at all, but sometimes I will write something like this:

```ruby
require 'concord'

# Example thing
class Thing
  include Concord.new(:foo, :bar)

  def initialize(foo)
    super(foo, foo * 100)
  end
end
```

which produces the following

```sh
$ reek reek-utility-initialize.rb
reek-utility-initialize.rb -- 1 warning:
  [7]:Thing#initialize doesn't depend on instance state (maybe move it to another class?) (UtilityFunction)
```

**Comment by backus** [comment]
@troessner is this helpful?

**Comment by troessner** [comment]
>> @troessner is this helpful?

It is!

>> I think this is sort of a duplicate of #39 except the "fix" for that issue only covered when zsuper is used. I think this should cover super() too.

I tend to agree. @mvz && @chastell what do you think?

**Comment by chastell** [comment]
Oh definitely!

**Comment by mvz** [comment]
Yes, I also agree.

**Comment by troessner** [comment]
I'm still busy with #724 - any of you guys available to look into this? If not, I can look into fixing this after #724

**Comment by mvz** [comment]
I can look into it.

**Comment by mvz** [comment]
Done. See #768.

**Comment by mvz** [comment]
Fixed in version 3.6.1!
